qualifier/src/main/scala/com/weshare/qualifier/ScenarioProcessor.scala

```diff
@@ -0,0 +1,94 @@
+package com.weshare.qualifier
+
+import com.twitter.util.Future
+import com.weshare.qualifier.ScenarioConfig._
+import org.jboss.netty.handler.codec.http._
+import org.slf4j.{Logger, LoggerFactory}
+
+class ScenarioProcessor extends DataContainers {
+  private val logger: Logger = LoggerFactory.getLogger(classOf[ScenarioProcessor])
+
+  //--------------------------------------------------------------------------------------------------------------------
+  //Below is scenario where user is not in an adventure
+  //--------------------------------------------------------------------------------------------------------------------
+
+  /**
+   * General workflow:
+   * 1. Create an adventure in the adventure table with the leader being the user id
+   * 2. Find the user's friends in the friends table and send then the invitation to join the adventure
+   * 3. Register a handler for each friend's response:
+   * - Acceptance + Do get previous photos    => Add user_id to adventure AND Send link to all previously taken photos in adventure
+   * - Acceptance + Don't get previous photos => Add user_id to adventure and send link to picture that triggered invite
+   * - Decline                                => Do Nothing
+   */
+  def createAdventureAndInviteFriends(requestData: RequestData) {
+    val boundary: Boundary = Boundary(requestData.geoTag)
+    val friendsInBoundary = for {
+      friend <- findFriendsOfUser(requestData.userId)
+      friendInBoundary <- boundary.withinBoundary(friend)
+    } yield friendInBoundary
+
+    logger.info(s"Sending an invitation to adventure: ${requestData.adventureToken} to users: $friendsInBoundary, with initial picture: ${requestData.pictureUrl}")
+    sendInvitationsToJoinAdventure(friendsInBoundary, requestData.adventureToken, requestData.pictureUrl)
+  }
+
+  def findFriendsOfUser(userId: String): List[(String, Geo)] = {
+    List(("2", (1.0, 2.0)), ("3", (3.0, 4.0)))
+  }
+
+  def sendInvitationsToJoinAdventure(friendsToInvite: List[String], adventureToken: String, pictureUrl: String) = {
+    friendsToInvite.map { friendToInvite =>
+      val routifierRequest = constructRequestJson(type_invitation, friendToInvite, pictureUrl, adventureToken)
+      logger.info(s"$routifierRequest")
+      handleInvitationResponse(qualifierClient(buildRoutifierRequest(type_invitation, friendToInvite, pictureUrl, adventureToken)), adventureToken, friendToInvite, pictureUrl)
+    }
+  }
+
+  def handleInvitationResponse(invitationResponse: Future[HttpResponse], adventureToken: String, friendInBoundary: String, pictureUrl: String): Future[HttpResponse] = {
+    invitationResponse.onSuccess { s =>
+      s.getStatus.getCode match {
+        case 200 =>
```

Comment:
Agree on the redundancy. However I am not sure they will stay the same, i.e. I am thinking about the differences between an invitation vs. sending picture and one may use a flatmap and return something different, instead of just passing along the original HttpResponse with OnSuccess. Also completely not critical just fyi, a non-200 http response code will not cause code to enter the OnFailure block (which has fucked up peeps at work). If the Future[HttpResponse] resolves with an exception you go to the OnFailure block, all http responses will be handled in the onSuccess block.